core/templates/services/audio-player.service.ts

```diff
@@ -16,162 +16,219 @@
-      if (_currentTrack) {
-        var currentSeconds = _currentTrack.progress * _currentTrack.duration;
-        if (currentSeconds - seconds > 0) {
-          var rewindedProgress = (
-            (currentSeconds - seconds) / _currentTrack.duration);
-          _currentTrack.progress = rewindedProgress;
-        } else {
-          _currentTrack.progress = 0;
-        }
-      }
-    };
-
-    var _forward = function(seconds) {
-      if (_currentTrack) {
-        var currentSeconds = _currentTrack.progress * _currentTrack.duration;
-        if (currentSeconds + seconds < _currentTrack.duration) {
-          var forwardedProgress = (
-            (currentSeconds + seconds) / _currentTrack.duration);
-          _currentTrack.progress = forwardedProgress;
-        }
-      }
-    };
-
-    return {
-      load: function(filename) {
-        return $q(function(resolve, reject) {
-          _load(filename, resolve, reject);
+import { EventEmitter, Injectable, NgZone } from '@angular/core';
+import { AudioFile } from 'domain/utilities/audio-file.model';
+import { AudioTranslationManagerService } from 'pages/exploration-player-page/services/audio-translation-manager.service';
+import { AssetsBackendApiService } from './assets-backend-api.service';
+import { ContextService } from './context.service';
+import { Howl } from 'howler';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { interval, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AudioPlayerService {
+  private _currentTrackFilename: string | null = null;
+  private _currentTrack: Howl | null = null;
+  private _lastPauseOrSeekPos: number | null = null;
+  private _loadingTrack = false;
+  private _updateViewEventEmitter = new EventEmitter<void>();
+  private _autoplayAudioEventEmitter = new EventEmitter<void>();
+  private _stopIntervalSubject = new Subject<void>();
+  constructor(
+    private assetsBackendApiService: AssetsBackendApiService,
+    private audioTranslationManagerService: AudioTranslationManagerService,
+    private contextService: ContextService,
+    private ngZone: NgZone
+  ) {}
+
+  private async _loadAsync(filename: string, successCallback, errorCallback) {
+    if (this._loadingTrack) {
+      throw new Error('Already loading a track... Please try again!');
+    }
+    if (this._currentTrackFilename === filename) {
+      return;
+    }
+    this.assetsBackendApiService.loadAudio(
+      this.contextService.getExplorationId(),
+      filename).then(
+      (loadedAudioFile: AudioFile) => {
+        this._currentTrack = new Howl({
+          src: [URL.createObjectURL(loadedAudioFile.data)],
+          format: ['mp3']
+        });
+        this._currentTrack.on('load', () => {
+          this._currentTrackFilename = loadedAudioFile.filename;
+          this._loadingTrack = false;
+          this._lastPauseOrSeekPos = 0;
+          this.play();
```

Comment:
Are we expecting to play the audio in the load call?

Reply:
I thought so... removed.